inst/stan/marginal_model/functions.stan

```diff
@@ -0,0 +1,10 @@
+real primarycensored_wrapper_lpmf(data int d, real mu, real sigma, data real pwindow) {
+  int dist_id = 1; // lognormal
+  array[2] real params;
+  params[1] = mu;
+  params[2] = sigma;
+  int d_upper = d + 1;
+  int primary_id = 1; // Uniform
+  array[0] real primary_params;
+  return primarycensored_lpmf(d | dist_id, params, pwindow, d_upper, positive_infinity(), primary_id, primary_params);
```

Comment:
For D you should be able to do it using the brms truncation support and get the same result as doing it via primarycensored (as long as it can find the lcdf function for a custom likelihood) or you might need to work around

Reply:
As in `primarycensored` doesn't support truncation? I thought it did? If it does then seems easiest to depend on it?